src/wp-admin/includes/template.php

```diff
@@ -2647,9 +2653,9 @@ function wp_star_rating( $args = array() ) {
 
 	$output  = '<div class="star-rating">';
 	$output .= '<span class="screen-reader-text">' . $title . '</span>';
-	$output .= str_repeat( '<div class="star star-full" aria-hidden="true"></div>', $full_stars );
-	$output .= str_repeat( '<div class="star star-half" aria-hidden="true"></div>', $half_stars );
-	$output .= str_repeat( '<div class="star star-empty" aria-hidden="true"></div>', $empty_stars );
+	$output .= str_repeat( '<div class="star star-full" aria-hidden="true"></div>', (int) $full_stars );
+	$output .= str_repeat( '<div class="star star-half" aria-hidden="true"></div>', (int) $half_stars );
+	$output .= str_repeat( '<div class="star star-empty" aria-hidden="true"></div>', (int) $empty_stars );
```

Comment:
`str_repeat` expects an integer, however, it does cast a float to integer in all <= PHP 8.0 and 8.1 (so far). See it in action here https://3v4l.org/VdA3O. 

No code change is needed here.

One change in this function is to document that PHP 8 fixes the float locale:

```php
// Non-English decimal places when the $rating is coming from a string.
$rating = (float) str_replace( ',', '.', $parsed_args['rating'] );
```

Reply:
Needs:  check if there are unhappy tests for this function to document behavior when different values and types are given. This will help to surface behavior for future PHP versions.